packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -356,26 +356,29 @@ abstract class TestRunner {
     this.serializer = new JsonProtoSerializer(this.databaseInfo.databaseId, {
       useProto3Json: true
     });
-    this.persistence = this.getPersistence(this.serializer);
+    this.persistence = new Promise(resolve => {
+      const p = this.getPersistence(this.serializer);
+      p.start().then(() => resolve(p));
+    });
```

Comment:
Hmm, this is somewhat strange. Can we not start the persistence layer in the renamed `start()`? We can check there whether it is already started using your new nifty API.

Reply:
We technically could, but `doRestart()` calls `start()`, and `doRestart()` intentionally does not want to start the persistence layer. As such, I think it makes more sense to keep the call to `persistence.start()` in the one place where we know we want to start it, rather than gate it with an `if (!persistence.started) await persistence.start()`, which then means you have to know why it would or wouldn't already be started.